woocommerce/woocommerce-gutenberg-products-block
# Fix handpicked product selections when a store has over 100 products.

There was a bug in `getProductsRequests` for stores with large catalogs. Our search functionality gets around page limits by including all selected products in a separate request. However, because this request had a limit or 10 products, some selected products would be left out of the list. Setting per_page to 0 (unlimited) resolves this.

When trying to make sense of the bug, I refactored the `withSearchedProducts` HOC to TypeScript.

Fixes #4505

### How to test the changes in this Pull Request:

To test this you need a catalog with > 100 products (or you can edit the code to make it require less).

1. Add a hand picked product block to a page.
2. Make 20 selections. You may need to search for products to have more results appear.
3. Save and reload the page.
4. Confirm 20 selections remain.

### Changelog

> Fix handpicked product selections when a store has over 100 products.
